presto-parser/src/main/java/com/facebook/presto/sql/tree/PathElement.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public final class PathElement
+        extends Node
+{
+    private final Optional<String> catalog;
+    private final String schema;
+
+    public PathElement(String schema)
```

Comment:
The current convention in presto is to minimize the amount of constructors, I see why this would be convenient but in general it's easier to reason about classes with a single constructor. Just use the Optional.empty() for missing arguments.